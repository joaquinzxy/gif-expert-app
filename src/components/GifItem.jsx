import PropTypes from 'prop-types';


export const GifItem = ({ url }) => {

    return (
    <a className="card" href={url}>
        <img src={url}/>
    </a>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}