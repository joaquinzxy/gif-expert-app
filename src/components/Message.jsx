export const Message = ({ message }) => {
  return (
    <div className='message-container'>
        <h3>{message}</h3>
    </div>
  )
}
