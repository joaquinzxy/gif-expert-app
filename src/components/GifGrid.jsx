import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import { PropTypes } from "prop-types";
import { useState } from "react";

export const GifGrid = ({ category, onDeleteCategory }) => {

  const gifsPerPage = 4;

  const { images, isLoading } = useFetchGifs(category);

  const [index, setIndex] = useState(0);

  const [paginationToggle, setPaginationToggle] = useState([true, false])

  const nextHandler = () => {
    setIndex(index + gifsPerPage);
    index < 36 ? setPaginationToggle([false, false]) : setPaginationToggle([false, true])
  }

  const previousHandler = () => {
    setIndex(index - gifsPerPage);
    index > gifsPerPage ? setPaginationToggle([false, false]) : setPaginationToggle([true, false])
  }

  return (
    <>
      <div className="gif-grid">
        <div className="gallery-header">
          <h4>▶️ {category}</h4>
          <div className="pagination">
            <button onClick={previousHandler} disabled={paginationToggle[0]}>←</button>
            <button onClick={nextHandler} disabled={paginationToggle[1]}>→</button>
            <button onClick={() => onDeleteCategory(category)} className={'close-btn'}> ✕ </button>
          </div>
        </div>
        {
          isLoading && (<h2>Cargando...</h2>)
        }
        <div className="card-grid">
          {
            images.slice(index, index + 4).map((image) => (
              <GifItem
                key={image.id}
                {...image} />
            ))
          }
        </div>
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
