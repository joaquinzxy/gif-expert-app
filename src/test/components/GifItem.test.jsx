import { GifItem } from "../../components";
import { render, screen } from "@testing-library/react";

describe('testing GifItem.jsx', () => { 
    
    const title = 'Saitama';
    const url = 'https://www.one-punch.com/saitama.jpg'

    test('should match with snapshot', () => { 


        const { container } = render(<GifItem title={ title } url={ url }/>)

        expect( container ).toMatchSnapshot();
     })

     test('should display img with url and alt provided', () => { 

        render(<GifItem title={ title } url={ url }/>);

        const { src, alt } = screen.getByRole( 'img' );
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

        // screen.debug();

      })

      test('should display title on component', () => { 

        render(<GifItem title={ title } url={ url }/>);

        expect(screen.getByText(title)).toBeTruthy();

       })

 })