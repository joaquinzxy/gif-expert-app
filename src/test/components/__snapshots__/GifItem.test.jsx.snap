// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifItem.jsx should match with snapshot 1`] = `
<div>
  <a
    class="card"
    href="https://www.one-punch.com/saitama.jpg"
  >
    <img
      alt="Saitama"
      src="https://www.one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </a>
</div>
`;
